### ISSUE TITLE
how to tell which packages would be upgraded by darwin-rebuild switch

### ISSUE THREAD

**Issue by mattsawyer77** [nix-darwin/nix-darwin#313]
Hello, I'm curious how to tell what will be upgraded by a `darwin-rebuild switch` operation? I've discovered I can do `darwin-rebuild check` and that seems to build all the new versions...but the output of that is quite verbose and it's not always easy to figure out the packages that it wants to upgrade. I have also tried executing `darwin-rebuild changelog` after updating my channels, but that never produces any output.

**Comment by eclairevoyant** [comment]
Use `nh` instead: https://github.com/nix-community/nh
It includes a diffing tool that compares current vs next generation.

**Comment by mattsawyer77** [comment]
@eclairevoyant thank you for the fantastic recommendation. `nh` seems to handle this question perfectly so far in addition to a lot of other useful functionality and perf improvements.

**Comment by Samasaur1** [comment]
Yep, that's out of scope for darwin-rebuild itself. You can do a `darwin-rebuild build` and then compare that to `/run/current-system` in whatever manner works for you — personally I have a hook set up to run `nvd diff`, but `nh` is also an option, and there are probably more out there as well.
